### PR TITLE
feat: support Stream and SSE in ExampleResponseSchema

### PR DESCRIPTION
**Pull request by abvthecity**
The response schema now accepts an undiscriminated union of body or stream, where stream itself can either be a list of data or a list of server side events, depending on the streaming format in the endpoint definition.

https://github.com/fern-api/fern/blob/1390e867dea2e89f728a0fa7c0ff8b7ebe49a9c8/packages/cli/yaml/yaml-schema/src/schemas/ExampleResponseSchema.ts#L4-L63

The fern check rule is updated to check that endpoint response-stream format (json or sse) matches the example response format, before also validating that the schema of the example aligns.

### CODE REVIEW

File: packages/cli/yaml/yaml-schema/src/schemas/ExampleResponseSchema.ts [hunk 1]
```diff
@@ -1,9 +1,73 @@
...
+    stream: z.array(ExamplSseEventSchema)
+});
+
+export type ExampleSseResponseSchema = z.infer<typeof ExampleSseResponseSchema>;
+
+export const ExampleResponseSchema = z.union([
+    ExampleBatchResponseSchema,
+    ExampleStreamResponseSchema,
+    ExampleSseResponseSchema
+]);
```
Comment (dsinghvi) [inline packages/cli/yaml/yaml-schema/src/schemas/ExampleResponseSchema.ts:71]: can we add an `sse` test defnition [here](https://github.com/fern-api/fern/tree/main/test-definitions/fern/apis) to make sure this works.

File: packages/cli/yaml/yaml-schema/src/schemas/ExampleResponseSchema.ts [hunk 2]
```diff
@@ -1,9 +1,73 @@
...
+ */
+export const ExampleStreamResponseSchema = z.strictObject({
+    stream: z.array(ExampleTypeReferenceSchema)
+});
+
+export type ExampleStreamResponseSchema = z.infer<typeof ExampleStreamResponseSchema>;
+
+export const ExamplSseEventSchema = z.strictObject({
+    event: z.string(),
+    data: z.optional(ExampleTypeReferenceSchema)
```
Comment (dsinghvi) [inline packages/cli/yaml/yaml-schema/src/schemas/ExampleResponseSchema.ts:41]: Can we update this `fern check` rule to check if the stream or data examples are actually valid? (https://github.com/fern-api/fern/tree/main/packages/cli/yaml/validator/src/rules/valid-example-endpoint-call)

File: packages/cli/yaml/validator/src/rules/valid-example-endpoint-call/validateResponse.ts [hunk 3]
```diff
@@ -26,15 +26,44 @@ export function validateResponse({
     file: FernFileContext;
     workspace: FernWorkspace;
     errorResolver: ErrorResolver;
+}): RuleViolation[] {
+    if (example == null) {
+        return [{ severity: "warning", message: "Response example is missing." }];
+    }
```
Comment (dsinghvi) [inline packages/cli/yaml/validator/src/rules/valid-example-endpoint-call/validateResponse.ts:32]: @abvthecity why did you add this?

Comment (abvthecity) [reply packages/cli/yaml/validator/src/rules/valid-example-endpoint-call/validateResponse.ts:32]: ah this is probably not needed now that i'm reading the original code.
`return [];`
